### ISSUE TITLE
redhat-openjdk18-openshift imagestream with binary BuildConfig does not work

### ISSUE THREAD

**Issue by arnaud-deprez** [jboss-openshift/application-templates#382]
Hi,

I'm not sure it's the right place to post this issue.

Following this blog post: https://blog.openshift.com/openshift-pipelines-jenkins-blue-ocean/
The s2i build from sources works correctly, but when you run the pipeline build it does not and it produce this error message in the jenkins build: 
```
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Build Image)
[Pipeline] unstash
[Pipeline] sh
[cicd-cart-service-pipeline] Running shell script
+ oc start-build cart --from-file=target/cart.jar --follow
Uploading file "target/cart.jar" as binary input for the build ...
build "cart-4" started
Receiving source from STDIN as file cart.jar
==================================================================
Starting S2I Java Build .....
S2I source build with plain binaries detected
Copying binaries from /tmp/src to /deployments ...
... done

Pushing image 172.30.1.1:5000/cicd/cart:latest ...
Pushed 5/6 layers, 84% complete
Pushed 6/6 layers, 100% complete
Push successful
Error from server (BadRequest): No field label conversion function found for version: build.openshift.io/v1
[Pipeline] }
[Pipeline] // stage
```

I'm wondering if that could be linked to: https://github.com/openshift/jenkins-sync-plugin/issues/173 but I'm using the latest jenkins image where it should be fixed.

I'm running on minishift: 
```
minishift start --openshift-version=v3.6.0
```

I tested it against redhat-openjdk18-openshift:1.0 and redhat-openjdk18-openshift:1.1 and it fails with both in the same way.

**Comment by tienhngnguyen** [comment]
I have the same problem when following that tutorial on my local OpenShift Origin running in a Docker container.

This is my environment:
Jenkins 2.73.2
Plugins:
OpenShift Pipeline Jenkins Plugin 1.0.52
OpenShift Sync 0.1.31

oc v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7
features: Basic-Auth

Server https://127.0.0.1:8443
openshift v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7

If I take a look into the logs of my Jenkins pod, I can see the following messages:

...
INFO: Waiting for Jenkins to be started
--
  | Oct 20, 2017 3:26:02 PM io.fabric8.jenkins.openshiftsync.BuildConfigWatcher start
  | INFO: Now handling startup build configs!!
  | Oct 20, 2017 3:26:02 PM io.fabric8.jenkins.openshiftsync.ConfigMapWatcher start
  | INFO: Now handling startup config maps!!
  | Oct 20, 2017 3:26:02 PM io.fabric8.jenkins.openshiftsync.ImageStreamWatcher start
  | INFO: Now handling startup image streams!!
  | Oct 20, 2017 3:26:02 PM org.springframework.context.support.AbstractApplicationContext prepareRefresh
  | INFO: Refreshing org.springframework.web.context.support.StaticWebApplicationContext@1538cca: display name [Root WebApplicationContext]; startup date [Fri Oct 20 15:26:02 UTC 2017]; root of context hierarchy
  | Oct 20, 2017 3:26:02 PM org.springframework.context.support.AbstractApplicationContext obtainFreshBeanFactory
  | INFO: Bean factory for application context [org.springframework.web.context.support.StaticWebApplicationContext@1538cca]: org.springframework.beans.factory.support.DefaultListableBeanFactory@1dc8eb5
  | Oct 20, 2017 3:26:02 PM org.springframework.beans.factory.support.DefaultListableBeanFactory preInstantiateSingletons
  | INFO: Pre-instantiating singletons in org.springframework.beans.factory.support.DefaultListableBeanFactory@1dc8eb5: defining beans [filter,legacy]; root of factory hierarchy
  | Oct 20, 2017 3:26:03 PM okhttp3.internal.platform.Platform log
  | INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
  | Oct 20, 2017 3:26:03 PM okhttp3.internal.platform.Platform log
  | INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
  | Oct 20, 2017 3:26:03 PM okhttp3.internal.platform.Platform log
  | INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
  | Oct 20, 2017 3:26:03 PM okhttp3.internal.platform.Platform log
  | INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
  | Oct 20, 2017 3:26:03 PM io.fabric8.jenkins.openshiftsync.ConfigMapWatcher$1 doRun
  | INFO: creating ConfigMap watch for namespace ci and resource version 8430
  | Oct 20, 2017 3:26:03 PM okhttp3.internal.platform.Platform log
  | INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
  | Oct 20, 2017 3:26:03 PM io.fabric8.jenkins.openshiftsync.BuildConfigWatcher$1 doRun
  | INFO: creating BuildConfig watch for namespace ci and resource version 8430
  | Oct 20, 2017 3:26:03 PM okhttp3.internal.platform.Platform log
  | INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
  | Oct 20, 2017 3:26:03 PM io.fabric8.jenkins.openshiftsync.BuildWatcher$1 doRun
  | INFO: creating Build watch for namespace ci and resource version 8430
  | Oct 20, 2017 3:26:03 PM hudson.WebAppMain$3 run
  | INFO: Jenkins is fully up and running
  | Oct 20, 2017 3:26:03 PM okhttp3.internal.platform.Platform log
  | INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
  | Oct 20, 2017 3:26:04 PM io.fabric8.jenkins.openshiftsync.ImageStreamWatcher$1 doRun
  | INFO: creating ImageStream watch for namespace ci and resource version 8430
  | Oct 20, 2017 3:26:05 PM okhttp3.internal.platform.Platform log
  | INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
  | Oct 20, 2017 3:26:07 PM org.openshift.jenkins.plugins.openshiftlogin.OpenShiftOAuth2SecurityRealm populateDefaults
  | INFO: OpenShift OAuth: provider: OpenShiftProviderInfo: issuer: https://127.0.0.1:8443 auth ep: https://127.0.0.1:8443/oauth/authorize token ep: https://127.0.0.1:8443/oauth/token
  | Oct 20, 2017 3:26:07 PM org.openshift.jenkins.plugins.openshiftlogin.OpenShiftOAuth2SecurityRealm populateDefaults
  | INFO: OpenShift OAuth returning true with namespace ci SA dir null default /run/secrets/kubernetes.io/serviceaccount SA name null default jenkins client ID null default system:serviceaccount:ci:jenkins secret null default eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJjaSIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VjcmV0Lm5hbWUiOiJqZW5raW5zLXRva2VuLXhtMWN4Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQubmFtZSI6ImplbmtpbnMiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC51aWQiOiI3NzE1ODQxNy1iNThjLTExZTctODU3NS0wZTRkZWZjODRiNDMiLCJzdWIiOiJzeXN0ZW06c2VydmljZWFjY291bnQ6Y2k6amVua2lucyJ9.TstDfIRNnKuiVV_SPB8kz9-I3t1XslIbjniahULDdmD4Z64v6C1ZY_WviauPqzta8oPLDt4mKm7XSw_6l1UtuFE_dIFaRhDgGR0rYoYsKrH4LOX979nnd0_zJa-COI4-Ew7yVHQwTGicQU9JSNg0cUlQRl4uxUf1IFkcaAiRfWKtGP3XAVHOEKNNHNaqVJ_i-zFHfPauFR4Y0nvxO3x3Qh3hsktt4bMihNoQSlNuEL7B7ktsITF942lRrSoXYGjKqu6hAh7vyZMs_c8ecaL25CVvyn8MunJsae1XmSRRO5Tvz42lwc5qJce3rOi3GWSGfMTwJleX4udMinFoi-7l2Q redirect null default https://127.0.0.1:8443 server null default https://openshift.default.svc
  | Oct 20, 2017 3:26:33 PM org.openshift.jenkins.plugins.openshiftlogin.OpenShiftOAuth2SecurityRealm populateDefaults
  | INFO: OpenShift OAuth: provider: OpenShiftProviderInfo: issuer: https://127.0.0.1:8443 auth ep: https://127.0.0.1:8443/oauth/authorize token ep: https://127.0.0.1:8443/oauth/token
  | Oct 20, 2017 3:26:33 PM org.openshift.jenkins.plugins.openshiftlogin.OpenShiftOAuth2SecurityRealm populateDefaults
  | INFO: OpenShift OAuth returning true with namespace ci SA dir null default /run/secrets/kubernetes.io/serviceaccount SA name null default jenkins client ID null default system:serviceaccount:ci:jenkins secret null default eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJjaSIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VjcmV0Lm5hbWUiOiJqZW5raW5zLXRva2VuLXhtMWN4Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQubmFtZSI6ImplbmtpbnMiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC51aWQiOiI3NzE1ODQxNy1iNThjLTExZTctODU3NS0wZTRkZWZjODRiNDMiLCJzdWIiOiJzeXN0ZW06c2VydmljZWFjY291bnQ6Y2k6amVua2lucyJ9.TstDfIRNnKuiVV_SPB8kz9-I3t1XslIbjniahULDdmD4Z64v6C1ZY_WviauPqzta8oPLDt4mKm7XSw_6l1UtuFE_dIFaRhDgGR0rYoYsKrH4LOX979nnd0_zJa-COI4-Ew7yVHQwTGicQU9JSNg0cUlQRl4uxUf1IFkcaAiRfWKtGP3XAVHOEKNNHNaqVJ_i-zFHfPauFR4Y0nvxO3x3Qh3hsktt4bMihNoQSlNuEL7B7ktsITF942lRrSoXYGjKqu6hAh7vyZMs_c8ecaL25CVvyn8MunJsae1XmSRRO5Tvz42lwc5qJce3rOi3GWSGfMTwJleX4udMinFoi-7l2Q redirect null default https://127.0.0.1:8443 server null default https://openshift.default.svc
  | Oct 20, 2017 3:26:33 PM org.openshift.jenkins.plugins.openshiftlogin.OpenShiftOAuth2SecurityRealm updateAuthorizationStrategy
  | INFO: OpenShift OAuth: user developer, stored in the matrix as developer-admin, based on OpenShift roles [view, edit, admin] already exists in Jenkins
  | Oct 20, 2017 3:27:10 PM io.fabric8.jenkins.openshiftsync.BuildConfigWatcher updateJob
  | INFO: Updated job ci-cart-service-pipeline from BuildConfig NamespaceName{ci:cart-service-pipeline} with revision: 8462
  | Oct 20, 2017 3:27:10 PM io.fabric8.jenkins.openshiftsync.BuildSyncRunListener onStarted
  | INFO: starting polling build job/ci-cart-service-pipeline/7/
  | Oct 20, 2017 3:27:42 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesCloud provision
  | INFO: Excess workload after pending Spot instances: 1
  | Oct 20, 2017 3:27:42 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesCloud provision
  | INFO: Template: Kubernetes Pod Template
  | Oct 20, 2017 3:27:42 PM okhttp3.internal.platform.Platform log
  | INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
  | Oct 20, 2017 3:27:43 PM hudson.slaves.NodeProvisioner$StandardStrategyImpl apply
  | INFO: Started provisioning Kubernetes Pod Template from openshift with 1 executors. Remaining excess workload: 0
  | Oct 20, 2017 3:27:43 PM okhttp3.internal.platform.Platform log
  | INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
  | Oct 20, 2017 3:27:43 PM org.csanchez.jenkins.plugins.kubernetes.ProvisioningCallback call
  | INFO: Created Pod: maven-xf7cn in namespace ci
  | Oct 20, 2017 3:27:43 PM org.csanchez.jenkins.plugins.kubernetes.ProvisioningCallback call
  | INFO: Waiting for Pod to be scheduled (0/100): maven-xf7cn
  | Oct 20, 2017 3:27:45 PM hudson.TcpSlaveAgentListener$ConnectionHandler run
  | INFO: Accepted JNLP4-connect connection #1 from /172.17.0.2:49622
  | Oct 20, 2017 3:27:49 PM okhttp3.internal.platform.Platform log
  | INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
  | Oct 20, 2017 3:27:52 PM hudson.slaves.NodeProvisioner$2 run
  | INFO: Kubernetes Pod Template provisioning successfully completed. We have now 2 computer(s)
  | Oct 20, 2017 3:29:39 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesSlave _terminate
  | INFO: Terminating Kubernetes instance for agent maven-xf7cn
  | Oct 20, 2017 3:29:39 PM org.jenkinsci.plugins.workflow.job.WorkflowRun finish
  | INFO: ci-cart-service-pipeline #7 completed: FAILURE
  | Oct 20, 2017 3:29:39 PM okhttp3.internal.platform.Platform log
  | INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
  | Oct 20, 2017 3:29:39 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesSlave _terminate
  | INFO: Terminated Kubernetes instance for agent ci/maven-xf7cn
  | Terminated Kubernetes instance for agent ci/maven-xf7cn
  | Oct 20, 2017 3:29:39 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesSlave _terminate
  | INFO: Disconnected computer maven-xf7cn
  | Oct 20, 2017 3:29:39 PM jenkins.slaves.DefaultJnlpSlaveReceiver channelClosed
  | WARNING: Computer.threadPoolForRemoting [#15] for maven-xf7cn terminated
  | java.nio.channels.ClosedChannelException
  | at org.jenkinsci.remoting.protocol.impl.ChannelApplicationLayer.onReadClosed(ChannelApplicationLayer.java:208)
  | at org.jenkinsci.remoting.protocol.ApplicationLayer.onRecvClosed(ApplicationLayer.java:222)
  | at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.onRecvClosed(ProtocolStack.java:832)
  | at org.jenkinsci.remoting.protocol.FilterLayer.onRecvClosed(FilterLayer.java:287)
  | at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.onRecvClosed(SSLEngineFilterLayer.java:181)
  | at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.switchToNoSecure(SSLEngineFilterLayer.java:283)
  | at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processWrite(SSLEngineFilterLayer.java:503)
  | at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processQueuedWrites(SSLEngineFilterLayer.java:248)
  | at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.doSend(SSLEngineFilterLayer.java:200)
  | at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.doCloseSend(SSLEngineFilterLayer.java:213)
  | at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.doCloseSend(ProtocolStack.java:800)
  | at org.jenkinsci.remoting.protocol.ApplicationLayer.doCloseWrite(ApplicationLayer.java:173)
  | at org.jenkinsci.remoting.protocol.impl.ChannelApplicationLayer$ByteBufferCommandTransport.closeWrite(ChannelApplicationLayer.java:311)
  | at hudson.remoting.Channel.close(Channel.java:1403)
  | at hudson.remoting.Channel.close(Channel.java:1356)
  | at hudson.slaves.SlaveComputer.closeChannel(SlaveComputer.java:708)
  | at hudson.slaves.SlaveComputer.access$800(SlaveComputer.java:96)
  | at hudson.slaves.SlaveComputer$3.run(SlaveComputer.java:626)
  | at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
  | at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
  | at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  | at java.lang.Thread.run(Thread.java:748)
  |  
  | Oct 20, 2017 3:29:39 PM hudson.remoting.Request$2 run
  | WARNING: Failed to send back a reply to the request hudson.remoting.Request$2@1078adc
  | hudson.remoting.ChannelClosedException: channel is already closed
  | at hudson.remoting.Channel.send(Channel.java:667)
  | at hudson.remoting.Request$2.run(Request.java:372)
  | at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
  | at org.jenkinsci.remoting.CallableDecorator.call(CallableDecorator.java:19)
  | at hudson.remoting.CallableDecoratorList$1.call(CallableDecoratorList.java:21)
  | at jenkins.util.ContextResettingExecutorService$2.call(ContextResettingExecutorService.java:46)
  | at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  | at java.lang.Thread.run(Thread.java:748)
  | Caused by: java.nio.channels.ClosedChannelException
  | at org.jenkinsci.remoting.protocol.impl.ChannelApplicationLayer.onReadClosed(ChannelApplicationLayer.java:208)
  | at org.jenkinsci.remoting.protocol.ApplicationLayer.onRecvClosed(ApplicationLayer.java:222)
  | at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.onRecvClosed(ProtocolStack.java:832)
  | at org.jenkinsci.remoting.protocol.FilterLayer.onRecvClosed(FilterLayer.java:287)
  | at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.onRecvClosed(SSLEngineFilterLayer.java:181)
  | at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.switchToNoSecure(SSLEngineFilterLayer.java:283)
  | at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processWrite(SSLEngineFilterLayer.java:503)
  | at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processQueuedWrites(SSLEngineFilterLayer.java:248)
  | at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.doSend(SSLEngineFilterLayer.java:200)
  | at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.doCloseSend(SSLEngineFilterLayer.java:213)
  | at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.doCloseSend(ProtocolStack.java:800)
  | at org.jenkinsci.remoting.protocol.ApplicationLayer.doCloseWrite(ApplicationLayer.java:173)
  | at org.jenkinsci.remoting.protocol.impl.ChannelApplicationLayer$ByteBufferCommandTransport.closeWrite(ChannelApplicationLayer.java:311)
  | at hudson.remoting.Channel.close(Channel.java:1403)
  | at hudson.remoting.Channel.close(Channel.java:1356)
  | at hudson.slaves.SlaveComputer.closeChannel(SlaveComputer.java:708)
  | at hudson.slaves.SlaveComputer.access$800(SlaveComputer.java:96)
  | at hudson.slaves.SlaveComputer$3.run(SlaveComputer.java:626)
  | at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
  | at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
  | ... 4 more
  |  
  | Oct 20, 2017 3:29:39 PM io.fabric8.jenkins.openshiftsync.BuildSyncRunListener onCompleted
  | INFO: onCompleted job/ci-cart-service-pipeline/7/
  | Oct 20, 2017 3:29:39 PM io.fabric8.jenkins.openshiftsync.BuildSyncRunListener onFinalized
  | INFO: onFinalized job/ci-cart-service-pipeline/7/

I would be very thankful if somebody could help in this matter.

**Comment by arnaud-deprez** [comment]
By removing the `--follow` argument, the build passed.

Basically, I replaced in the pipeline:
`oc start-build cart --from-file=target/cart.jar --follow`
by
```
oc start-build cart --from-file=target/cart.jar
openshiftVerifyBuild bldCfg: "cart", waitTime: "20", waitUnit: "min"
```

So I think the issue is related to the `oc` client inside the jenkins slave.

**Comment by bparees** [comment]
@arnaud-deprez i think it might be a known issue w/ certain levels of the oc client binary plus certain versions of the openshift cluster.

@deads2k, @csrwng and @gabemontero know the details but I don't think we need a new issue for it.

Sounds like you have a workaround but the other option would be to move back to the jenkins v3.6 image which has an older oc client in it:

https://hub.docker.com/r/openshift/jenkins-2-centos7/tags/
https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/openshift3/jenkins-2-rhel7

Alternatively i believe this is being fixed in a future 3.6.x release for OCP (but if you're on origin i don't think we'll be delivering a patch for the 3.6.x stream)

**Comment by arnaud-deprez** [comment]
Ok thank you for the informations!
